File: tests/integration/languages/java/type-vocabulary.java
// Principle #14: type references wrap their name in a <name> child.
// Covers: plain types, generic types, array types, extends/implements,
// type parameter declarations with <extends> bounds.

import java.util.List;

class Animal {}
interface Barker { void bark(); }
interface Runner { void run(); }

class Dog<T extends Animal> extends Animal implements Barker, Runner {
    T owner;
    List<String> tags;

    public void bark() {}
    public void run() {}
}

File: tests/integration/languages/rust/type-vocabulary.rs
// Principle #14: type references wrap their name in a <name> child.
// Covers: primitive types, paths, generic types, reference types,
// type parameter declarations with <extends> bounds.

use std::collections::HashMap;

trait Barker { fn bark(&self); }

struct Dog<T: Barker> {
    owner: T,
    tags: Vec<String>,
    scores: HashMap<String, i32>,
    parent: Option<Box<Dog<T>>>,
}

impl<T: Barker> Barker for Dog<T> {
    fn bark(&self) {}
}
